src/transform/globalize.js

```diff
@@ -29,30 +28,93 @@ function indent (code) {
   return lines.join('\n');
 }
 
-function relative (file) {
+function makePathRelative (file) {
   return path.relative(process.cwd(), file);
 }
 
+function defineDependencies (imports, dependencies) {
+  var code = '';
+  var keyVals = imports.map(function (imp, idx) {
+    return '"' + imp + '": ' + generateModuleName(dependencies[idx]);
+  });
+
+  keyVals.unshift('"exports": exports');
+  keyVals.unshift('"module": module');
+  code = '\n' + indent(keyVals.join(',\n')) + '\n';
+  return '{' + code + '}';
+}
+
+function defineReplacement (name, deps, func) {
+  function overload (type) {
+    var typefn = type;
+
+    if (typeof typefn === 'string') {
+      typefn = function (ref) {
+        return type === 'array' ? Array.isArray(ref) : typeof ref === type;
+      };
+    }
+
+    return function () {
+      return [].slice.call(arguments).filter(function (value) {
+        return typefn(value);
+      })[0];
+    };
+  }
+
+  func = overload('function')(func, deps, name);
+  deps = overload('array')(deps, name, []);
+  rval = func.apply(null, deps.map(function (value) {
+    return defineDependencies[value];
+  }));
+
+  if (rval) {
+    exports = module.exports = rval;
+  }
+}
+
 module.exports = function () {
   return function (data, info) {
-    var sourcemap = data.match(regexSourcemap);
```

Comment:
This was leaked from a previous commit. Didn't affect the code, but removing because it will be implemented in a separate PR.